### PR TITLE
Implement constant pool writing

### PR DESCRIPTION
**Pull request by rkennke**
Implement basic writing of constant pool items. This only implements and fixes cpool writing so much that it can correctly emit a parsable recording. Further implementation requires trace-ID infrastructure, which comes with a separate PR.

### CODE REVIEW

File: substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrChunkWriter.java [hunk 1]
```diff
@@ -312,6 +330,33 @@ public void writeCompressedLong(long value) throws IOException {
...
+        UTF8_BYTE_ARRAY(3),
+        CHAR_ARRAY(4),
+        LATIN1_BYTE_ARRAY(5);
+        private byte byteValue;
+        StringEncoding(int byteValue) {
+            this.byteValue = (byte) byteValue;
+        }
+    }
+
+    private static final Charset UTF8 = Charset.forName("UTF-8");
```
Comment (christianhaeubl) [inline substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrChunkWriter.java]: `StandardCharsets.UTF_8` could be used instead.

File: substratevm/src/com.oracle.svm.test/src/com/oracle/svm/test/jfr/JFR.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2020, 2021, Red Hat, Inc.
```
Comment (christianhaeubl) [inline substratevm/src/com.oracle.svm.test/src/com/oracle/svm/test/jfr/JFR.java]: I think that those files will need a different copyright header.

**Review by christianhaeubl — APPROVED** [review]

File: substratevm/src/com.oracle.svm.test/src/com/oracle/svm/test/jfr/JFR.java (hunk 2 above)
Comment (rkennke) [reply substratevm/src/com.oracle.svm.test/src/com/oracle/svm/test/jfr/JFR.java]: What do you suggest / don't you like? I believe it's what we use elsewhere for new files.

Comment (christianhaeubl) [reply substratevm/src/com.oracle.svm.test/src/com/oracle/svm/test/jfr/JFR.java]: The gate checks the copyright header (e.g., each copyright header must mention Oracle as well). Here is one compliant example that @adinn used for a file that he added in the context of his debugging info support: https://github.com/adinn/graal/blob/2c3c8abf4c59df5492f9add84f8d5b442c56825d/substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/ArrayTypeEntry.java

**Review by jiekang — APPROVED** [review]
Looks fine.

**Comment by rkennke** [comment]
@christianhaeubl GH actions complain that the new tests don't work. The tests (and JFR in general) currently require -H:+AllowVMInspection in order to make them work. Is it possible to make those tests pick up that flag/setting automatically so that GH actions can run them? Or what would be the way out of this?

**Pull request merged**